### PR TITLE
[Merged by Bors] - feat: Support podOverrides

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Part of https://github.com/stackabletech/issues/issues/346
Supersedes #433

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases added
- [x] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Comment by sbernauer** [comment]
Started https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/trino-operator-it-custom/80/

File: tests/templates/kuttl/resources/10-assert.yaml.j2 [hunk 1]
```diff
@@ -104,3 +104,37 @@ spec:
   resources:
     requests:
       storage: 4Gi
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: trino-worker-resources-from-role
```
Comment (adwk67) [inline tests/templates/kuttl/resources/10-assert.yaml.j2:111]:
```suggestion
  name: trino-worker-resources-from-pod-overrides
```

File: tests/templates/kuttl/resources/10-assert.yaml.j2 [hunk 2]
```diff
@@ -104,3 +104,37 @@ spec:
...
+        - name: vector
+{% endif %}
+status:
+  readyReplicas: 1
+  replicas: 1
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: data-trino-worker-resources-from-role-0
```
Comment (adwk67) [inline tests/templates/kuttl/resources/10-assert.yaml.j2:134]:
```suggestion
  name: data-trino-worker-resources-from-pod-overrides-0
```

**Review by adwk67 — APPROVED** [review]
lgtm % minor typos

**Comment by sbernauer** [comment]
bors r+
Resources test failed, fixed it and passed locally

**Pull request closed without merging**